packages/common/src/worker-deployments.ts

```diff
@@ -0,0 +1,59 @@
+import type { temporal } from '@temporalio/proto';
+import { makeProtoEnumConverters } from './internal-workflow';
+
+/**
+ * Represents the version of a specific worker deployment.
+ *
+ * @experimental Deployment based versioning is experimental and may change in the future.
+ */
+export interface WorkerDeploymentVersion {
+  readonly buildId: string;
+  readonly deploymentName: string;
+}
+
+/**
+ * @returns The canonical representation of a deployment version, which is a string in the format
+ * `deploymentName.buildId`.
+ */
```

Comment:
Please add `@internal` and `@hidden`.

Reply:
This isn't really either of those things, it can show up in the UI that way, etc. As of the latest API changes now that's less of a thing since this is actually well typed everywhere - but I don't know that we need to hide this really. We didn't in the other SDKs.

Reply:
In that case, let's at least give it a more specific name. This will be exported from the `common` package, and can't be renamed in the future. Maybe `toCanonicalDeploymentName`?